### PR TITLE
Update CAPI to v1.4.3

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Updates CAPI references to use the [v1.4.3 release](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.4.3). Also bumps cert-manager to v1.12.1 to [match CAPI](https://github.com/kubernetes-sigs/cluster-api/pull/8748). Reimplements some trivial string processing methods that were deprecated in CAPI.

**Which issue(s) this PR fixes**:

Closes #3613 

**Special notes for your reviewer**:

- [x] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update CAPI to v1.4.3
```

### CODE REVIEW

File: azure/scope/machine.go
Comment (CecileRobertMichon) [inline azure/scope/machine.go:461]: I wonder if we really need to do the parsing here or if we should just return the string as is. Since CAPI has gotten rid of Provider ID parsing altogether. And we're also not returning any error here if the parsing fails so this isn't adding any validation value.

File: azure/scope/machine.go
Comment (CecileRobertMichon) [inline azure/scope/machine.go:456]: same thing here

File: azure/scope/machine.go
Comment (mboersma) [reply azure/scope/machine.go:461]: It passes it through regex validation, but doesn't actually return an error if it doesn't match (following the existing contract of these funcs). But you could argue the `ProviderID()` func should be a no-op. I'll try that.

File: azure/scope/machine.go
Comment (CecileRobertMichon) [reply azure/scope/machine.go:456]: For this one we need to extract the VM ID so we do need to parse it. Could we use https://github.com/Azure/azure-sdk-for-go/blob/a688cfe08303253b05efb19a7645fbdb24d7375e/sdk/azcore/arm/resource_identifier.go#L20 instead of writing our own parsing function?

File: azure/scope/machine.go
Comment (mboersma) [reply azure/scope/machine.go:461]: We do have a MachineScope test that expects its `ProviderID()` func to do validation here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/machine_test.go#L198

IDK if the unit test was just written to match the (unneeded) parsing functionality or not, but it's a simple change if we're ok removing that test case.

**Comment by mboersma** [comment]
/hold

Let's rebase this after #3616 merges, so it doesn't duplicate code.

File: azure/scope/machinepool.go
Comment (CecileRobertMichon) [inline azure/scope/machinepool.go:161]: is resourceID.Name() the equivalent of what we had before here? Seems that we're returning the instance ID (which itself seems strange since the function name implies it returns the provider ID of the resource.

Let's add a small unit test that passes before the change to validate before and after is equivalent?

Comment (mboersma) [reply azure/scope/machinepool.go:161]: Comparing the code behind each, `resourceID.Name` should be equivalent, but I agree it deserves a unit test. Will do.

Comment (mboersma) [reply azure/scope/machinepool.go:161]: I added a unit test for `MachinePoolScope.ProviderID()` against the `main` branch, then cherry-picked it into this branch to see that it continues to pass after the reimplementation. 👍🏻

File: azure/scope/machinepool_test.go
Comment (CecileRobertMichon) [inline azure/scope/machinepool_test.go:113]: can we add a unit test with a VMSS provider ID? Would be good to have both coverage for both uniform and flex mode

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: azure/scope/machinepool_test.go
Comment (mboersma) [reply azure/scope/machinepool_test.go:113]: Sure, good idea.

Comment (mboersma) [reply azure/scope/machinepool_test.go]: Updated with test cases specifically for VMSS Uniform and Flex strings.

**Pull request merged**
